hashmap.c

```diff
@@ -50,6 +50,23 @@ unsigned int memihash(const void *buf, size_t len)
 	return hash;
 }
 
+/*
+ * Incoporate another chunk of data into a memihash
+ * computation.
+ */ 
+unsigned int memihash2(unsigned int hash_seed, const void *buf, size_t len)
+{
+	unsigned int hash = hash_seed;
+	unsigned char *ucbuf = (unsigned char *) buf;
+	while (len--) {
+		unsigned int c = *ucbuf++;
+		if (c >= 'a' && c <= 'z')
```

Comment:
@IDisposable  Thanks for the question.

Yes, that is correct.  My PR here was limited to reducing duplicated computations for performance reasons.  The caller was computing the hash on the full path and then on the full path of the parent directory.  We can go faster if we compute the hash on the parent directory first and then use that value to "continue" the computation using the filename portion.

My memihash2() (later memihash_cont()) just mirrors the implementation in the original memihash() and strihash() functions.  I did not address the correctness (or limitations) of the original computation.

Without soapboxing too much, yes, addressing non-US-ASCII case issues is a much larger problem in the git source tree (and for any cross-platform version control system).   Besides the ('a' - 'A') hack here, even calls to strcasecmp() are wrong/insufficient (and for the same reason (since it does the same trick under the hood)).  Even converting to a UTF-8-aware case-insensitive string compare routine is not sufficient -- it may help, but it doesn't completely address the problem.

What we really need is a way to establish "path equivalence" (and predict "path collisions").  The hash maps here are attempting to do that -- but only by considering US-ASCII -- a true/correct solution is much more difficult.  I don't blame them, these problems are made very difficult by the variations allowed by the various file systems we're given.  For example, Apple and HFS+ have their own rules for case and Unicode NFC/NFD, Windows has per-OS-version-specific case folding tables, Windows has final space and dot issues, Windows 8.3 aliasing, and etc.

For example, here is a test case I proposed for LibGit2 to explore properly fixing case-insensitivity on Windows:
https://github.com/libgit2/libgit2/pull/3191